synapse/storage/database.py

```diff
@@ -175,7 +175,7 @@ def commit(self) -> None:
     def rollback(self) -> None:
         self.conn.rollback()
 
-    def __enter__(self) -> "Connection":
+    def __enter__(self) -> "LoggingDatabaseConnection":
```

Comment:
The return value here makes its way into the data store constructor:
https://github.com/matrix-org/synapse/blob/3b8872299aac25a7e3ee5a9e00564105aa6de237/synapse/storage/databases/__init__.py#L64
https://github.com/matrix-org/synapse/blob/3b8872299aac25a7e3ee5a9e00564105aa6de237/synapse/storage/databases/__init__.py#L92